### DISCUSSION TITLE
Why datepicker cannot custom quick options

### DISCUSSION THREAD

**Discussion by Jingxi-Su** [primefaces/community#422] [PrimeVue]
I saw date picker has a prop "showButtonBar" to show Taday&Clear at the bottom. Why doesn't it have quick options like last 2/5/7 days, last month, .etc on the left? Maybe it's a good idea to add it.

**Comment by tugcekucukoglu** [comment]
Thanks for the feedback!

`header` and `footer` slots can be used as a quick solution.

**Comment by bobby-driggs** [comment]
Came here to also suggest this. Based on this github issue, it seems like it was suggested 2 years ago as well: https://github.com/primefaces/primevue/issues/2870

would love to have a customizeable button bar. For now were probably going to use the footer and either hack some CSS together, or i dunno, maybe use Vue's teleport feature to append it to the buttonbar class

**Reply by bobby-driggs** [reply]
In case anyone finds this later, here was my solution to this problem: Basically make a wrapping component, and when the div get created for the date picker panel, create a button via `h` and shove it into the buttonbar programatically, create your own `done` or whatever button named handler / event to emit, and wash your hands of it

```
<template>
  <DatePicker ref="datePicker" @show="onShow" />
</template>

<script setup lang="ts">
import { getCurrentInstance, h, onMounted, ref, render } from 'vue'
import type { ComponentInternalInstance } from 'vue'

import Button from 'primevue/button'
import DatePicker from 'primevue/datepicker'

let currentInstance: ComponentInternalInstance | null

const datePicker = ref<InstanceType<typeof DatePicker>>()

onMounted(() => {
  currentInstance = getCurrentInstance()
})

const props = defineProps({
  showClosedButton: {
    type: Boolean,
    default: true
  }
})

const emit = defineEmits(['done'])

const onShow = () => {
  if (props.showClosedButton !== true) {
    return
  }

  // virtually create primevue button that will close date panel when clicked
  const doneBtn = h(Button, {
    label: 'Done',
    outlined: true,
    size: 'small',
    onClick: () => {
      if (!!datePicker.value && !!datePicker?.value) {
        datePicker.value.onButtonClick()
        emit('done')
      }
    }
  })

  const instanceElId: string = currentInstance?.vnode?.el?.id
  const panelId = `${instanceElId}_panel`

  const panelEl = document.querySelector(`#${panelId}`)
  const buttonBarEl = panelEl?.querySelector('.p-datepicker-buttonbar')

  // appends done button to the button bar
  render(doneBtn, buttonBarEl!)
}
</script>
```
which nets:

![image](https://github.com/user-attachments/assets/06b29692-ffcb-4ff3-baf2-d987debf293e)

we didnt want to use `footer` slot, because it would have put the button below the button bar (the today / clear). And if i have that disabled, i would have to essentially reimplmeent those buttons.

**Reply by rowanpreesman** [reply]
This was **exactly** what I was looking for thanks!

In honesty, what you have made should be the standard `showButtonbar`. Users expect a bottom-right option to be something like `next`, `close` or `save` as this is the most common UX pattern for modals, steppers, popovers etc. Placing a reset action on that spot is a bit evil, and as an option itself has been subject to debate since dawn (https://www.nngroup.com/articles/reset-and-cancel-buttons/).

PrimeTek team, would it please be possible to adopt @bobby-driggs solution as standard for the datepicker `showButtonbar`? You can name the button `Close` and everyone will be happy :)
